docs/profiling.md

```diff
@@ -0,0 +1,108 @@
+# vHive profiling tool
+
+The tool focus on the performance characteristics of vHive framework and identify potential 
+performance bottleneck of colocating VMs on a single host. The motivation of this tool is to 
+improve the parallelism of serverless architecture and commercial benefits for cloud vendors.
+
+## Methodology
+
+The tool is for benchmarking vHive instances by recording hardware counters, requests per second
+ (RPS) and execution time for functions running inside instances.
+
+The tool includes the following components:
+- A loader function issues requests to vHive step by step until the RPS reaches the maximum. At 
+  each step, the function issues a percentage of the maximum RPS (e.g., 5% of the maximum RPS) 
+  and profile counters. If tail latency violates 10x image unloaded service time at a step, the 
+  function returns the metric before it.
+- A single latency measurement goroutine measures the latency by injecting requests in 
+  Round-Robin and measures runtimes of requests to compute the mean latency and the 
+  90-percentile latency.
+- The tool can bind only 1 VM to a core for profiling and other VMs to a socket
+- A profiler invokes [toplev](https://github.com/andikleen/pmu-tools) to collect hardware counters.
+- A plotter plots line charts. The X-axis is the number of VMs and Y-axis is the value of a metric.
+
+A load step is divided into three phases: warm-up, profiling and cool-down. During the profile 
+period, the benchmark records the average execution time of invocations and how many invocations 
+return successfully. During the profile period, it also invokes [pmu-tools](https://github.com/andikleen/pmu-tools) 
+to profile user-defined counters followed by [the TopDown method](https://ieeexplore.ieee.org/document/6844459). 
+After the measurement finishes, completed RPS per core, average execution time and the average 
+counters are saved in the `profile.csv`.
+
+## Runtime Arguments
+```
+General:
+-warmUpTime   FLOAT  The warm up time before profiling in seconds (default 5)
+-profileTime  FLOAT  The profiling time in seconds (default 10)
+-coolDownTime FLOAT  The cool down time after profiling in seconds (default 1)
+-loadStep     FLOAT  The percentage of target RPS the benchmark loads at every step (default 5)
+-funcNames    STR    Names of the functions to benchmark, separated by comma (default helloworld)
+-profileCPU   INT    Bind only one VM to the CPU and profile the physical core only (default -1)
+-bindSocket   INT    Bind all VMs to socket number apart from the profile CPU (default -1)
+-latSamples   INT    The number of latency measurements during one profiling period (default 100)
+
+TestProfileSingleConfiguration:
```

Comment:
Suggestion: TestProfileSingleConfiguration -> TestProfileFixedVMs

Reply:
Dmitrii thinks fixed VMs is not clear, so I changed to single configuration. The same reason for increment configuration